src/main/java/org/jenkinsci/remoting/engine/JnlpProtocolHandlerFactory.java

```diff
@@ -92,7 +87,9 @@ public JnlpProtocolHandlerFactory(@Nonnull ExecutorService threadPool) {
      * @return {@code this} for method chaining.
      */
     public JnlpProtocolHandlerFactory withNioChannelHub(@CheckForNull NioChannelHub nioChannelHub) {
-        this.nioChannelHub = nioChannelHub;
+        /**
+         * The {@link NioChannelHub} to use or {@code null}
+         */
```

Comment:
Is this supposed to be removed with no effect?

Reply:
@res0nance, as Matt comments, the nioChannelHub field isn't used. I haven't dug into the details of when this might have been used, though I'm not certain it ever was. The value is set on this line, but nothing ever uses it. As part of this PR, I've retained the public API, but removed the unused internal pieces.